File: java/src/main/java/com/experian/dvs/client/address/layout/attributes/GBRHealth.java
package com.experian.dvs.client.address.layout.attributes;

import java.util.Arrays;
import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public enum GBRHealth {

    AUTHORITY_CODE("authority_code"),
    AUTHORITY_CODE_2011("authority_code_2011"),
    AUTHORITY_NAME("authority_name"),
    PCLH_CODE("pclh_code"),
    PCLH_CODE_2011("pclh_code_2011"),
    PCLH_NAME("pclh_name"),
    WARD_CODE("ward_code"),
    WARD_CODE_2011("ward_code_2011"),
    WARD_NAME("ward_name"),
    CCG_CODE("ccg_code"),
    CCG_NAME("ccg_name"),
    DOH_CODE("doh_code");

    private final String value;
    // Map of value to GBRHealth
    private static final Map<String, GBRHealth> valueMap = Arrays.stream(GBRHealth.values()).collect(Collectors.toMap(GBRHealth::getValue, Function.identity()));

    public static GBRHealth fromValue(final String value) {
        return valueMap.get(value);
    }

    GBRHealth(final String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: typescript/src/address/format/enrichment/gbrHealth.ts
import { RestApiEnrichmentGbrHealth } from "../../../server/address/format/restApiEnrichmentGbrHealth";

export type GbrHealth = {
    authorityCode: string;
    authorityCode2011: string;
    authorityName: string;
    pclhCode: string;
    pclhCode2011: string;
    pclhName: string;
    wardCode: string;
    wardCode2011: string;
    wardName: string;
    ccgCode: string;
    ccgName: string;
    dohCode: string;
};

export function restApiResponseToGbrHealth(response: RestApiEnrichmentGbrHealth): GbrHealth {
    return {
        authorityCode: response.authority_code ?? "",
        authorityCode2011: response.authority_code_2011 ?? "",
        authorityName: response.authority_name ?? "",
        pclhCode: response.pclh_code ?? "",
        pclhCode2011: response.pclh_code_2011 ?? "",
        pclhName: response.pclh_name ?? "",
        wardCode: response.ward_code ?? "",
        wardCode2011: response.ward_code_2011 ?? "",
        wardName: response.ward_name ?? "",
        ccgCode: response.ccg_code ?? "",
        ccgName: response.ccg_name ?? "",
        dohCode: response.doh_code ?? ""
    };
}